modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -92,67 +96,57 @@ abstract class HealthCheckConsensus[
   private def createRoundId: F[RoundId] = GenUUID[F].make.map(RoundId.apply)
 
   def startOwnRound(key: HealthCheckKey) =
-    clusterStorage.getPeers.flatMap { peers =>
-      createRoundId.map(HealthCheckRoundId(_, selfId)).flatMap {
-        startRound(key, peers, _, OwnRound)
-      }
+    createRoundId.map(HealthCheckRoundId(_, selfId)).flatMap {
+      startRound(key, _)
     }
 
   def participateInRound(key: HealthCheckKey, roundId: HealthCheckRoundId): F[Unit] =
-    clusterStorage.getPeers.flatMap { peers =>
-      startRound(key, peers, roundId, PeerRound)
-    }
+    startRound(key, roundId)
 
   def startRound(
     key: HealthCheckKey,
-    peers: Set[Peer],
-    roundId: HealthCheckRoundId,
-    roundType: HealthCheckRoundType
+    roundId: HealthCheckRoundId
   ): F[Unit] =
-    Clock[F].monotonic
-      .map(_.toSeconds)
-      .flatMap { startedAt =>
-        allRounds.get.map {
-          case ConsensusRounds(historical, inProgress) =>
-            def inProgressRound = inProgress.get(key)
-            def historicalRound = historical.find(_.roundId == roundId)
-
-            historicalRound.orElse(inProgressRound).map(_ => none).getOrElse {
-              HealthCheckRound.make[F, A, B]().some
+    allRounds.get.map {
+      case ConsensusRounds(historical, inProgress) =>
+        def inProgressRound = inProgress.get(key)
+        def historicalRound = historical.find(_.roundId == roundId)
+
+        historicalRound.orElse(inProgressRound).map(_ => none).getOrElse {
```

Comment:
.map(_ => none)

No matter if you have historicalRound or inProgressRound 🤔 

Then getOrElse always triggers else part.

Reply:
No, it's `map` not `flatMap`. Option is nested

Reply:
Ah right

Reply:
tl;dr;
if there is any historical or in progress round -> return `None`
if there is no round, create one and return `Some(newRound)`